File: algorithm_basic/data_structure/editor/editor.py
from sys import stdin
class Editor:
    def __init__(self, _string):
        self.left_cursor = _string
        self.right_cursor = []
    def cmd_L(self):
        if len(self.left_cursor):
            ch = self.left_cursor.pop()
            self.right_cursor.append(ch)
    def cmd_D(self):
        if len(self.right_cursor):
            ch = self.right_cursor.pop()
            self.left_cursor.append(ch)
    def cmd_B(self):
        if len(self.left_cursor):
            self.left_cursor.pop()
    def cmd_P(self, ch):
        self.left_cursor.append(ch)
    def cmd_print(self):
        return "".join(self.left_cursor+self.right_cursor[::-1])

string = Editor(list(stdin.readline().strip()))
cmd_cnt = int(stdin.readline())
for i in range(cmd_cnt):
    cmd = stdin.readline()
    if cmd[0] == 'L':
        string.cmd_L()
    elif cmd[0] == 'D':
        string.cmd_D()
    elif cmd[0] == 'B':
        string.cmd_B()
    else:
        string.cmd_P(cmd[2])
print(string.cmd_print())

File: algorithm_basic/data_structure/editor/editor.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

class Editor {
	private:
		stack<char> left_cursor;
		stack<char> right_cursor;
		string result = "";
	public:
		Editor(string sentence) {
			for(int i = 0; i < sentence.size(); i++) {
				left_cursor.push(sentence[i]);
			}
		}
		void cmd_P(char element) {
			left_cursor.push(element);
		}
		void cmd_L() {
			if(left_cursor.size()){
				right_cursor.push(left_cursor.top());
				left_cursor.pop();
			}
		}
		void cmd_D() {
			if(right_cursor.size()){
				left_cursor.push(right_cursor.top());
				right_cursor.pop();
			}
		}
		void cmd_B() {
			if(left_cursor.size())
				left_cursor.pop();
		}
		string cmd_Print() {
			while(!left_cursor.empty()) {
				right_cursor.push(left_cursor.top());
				left_cursor.pop();
			}
			while(!right_cursor.empty()) {
				result += right_cursor.top();
				right_cursor.pop();
			}
			return result;
		}
};

int main() {
	string sentence;
	char cmd;
	int cnt_cmd;
	cin >> sentence >> cnt_cmd;
	Editor myEditor(sentence);
	for(int i = 0; i < cnt_cmd; i++) {
		cin >> cmd;
		switch(cmd) {
			case 'P': {
				char element;
				cin >> element;
				myEditor.cmd_P(element);
				break;
			}
			case 'L':
				myEditor.cmd_L();
				break;
			case 'D':
				myEditor.cmd_D();
				break;
			case 'B':
				myEditor.cmd_B();
				break;
			default:
				cout << "invalid option" << endl;
		}
	}
	cout << myEditor.cmd_Print();
	return 0;
}